contracts/src/v0.8/dev/ocr2dr/OCR2DRClient.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import "./OCR2DR.sol";
+import "../interfaces/OCR2DRClientInterface.sol";
+import "../interfaces/OCR2DROracleInterface.sol";
+
+/**
+ * @title The OCR2DR client contract
+ * @dev This contract is similar to ChainlinkClient in many respects.
+ * @notice Contract writers can inherit this contract in order to create on-demand OCR requests
+ */
+abstract contract OCR2DRClient is OCR2DRClientInterface {
+  OCR2DROracleInterface private s_oracle;
+  uint256 private s_requestCount = 1;
+  mapping(bytes32 => address) private s_pendingRequests;
+
+  event RequestSent(bytes32 indexed id);
+  event RequestFulfilled(bytes32 indexed id);
+  event RequestCancelled(bytes32 indexed id);
+
+  error TransferAndCallError();
```

Comment:
We're not going to use TransferAndCall, are we?

Reply:
We agreed on a different billing model. Therefore we are not going to transfer any value.